e2e_tests/integrations/test_huggingface.py

```diff
@@ -0,0 +1,163 @@
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+import numpy as np
+import pytest
+import torch
+from faker import Faker
+from transformers import Trainer, TrainingArguments
+from transformers.integrations import NeptuneCallback
+from transformers.utils import logging
+
+from e2e_tests.base import BaseE2ETest
+from e2e_tests.utils import catch_time, modified_environ
+from neptune.new import get_project, init_run
+
+EXPECTED_OVERWHELMING_FACTOR = 1.2
+
+
+logging.set_verbosity_error()
+fake = Faker()
+
+
+class RegressionModel(torch.nn.Module):
+    def __init__(self, initial_a=0, initial_b=0):
+        super().__init__()
+
+        # pylint: disable=no-member
+        self.a = torch.nn.Parameter(torch.tensor(initial_a).float())
+        self.b = torch.nn.Parameter(torch.tensor(initial_b).float())
+
+        self.config = None
+
+    def forward(self, input_x, labels=None):
+        y = input_x * self.a + self.b
+
+        if labels is None:
+            return (y,)
+
+        loss = torch.nn.functional.mse_loss(y, labels)
+
+        return loss, y
+
+
+class RegressionDataset:
+    def __init__(self, a=2, b=3, length=64, seed=2501):
+        np.random.seed(seed)
+
+        self.label_names = ["labels"]
+        self.length = length
+        self.x = np.random.normal(size=(length,)).astype(np.float32)
+        self.ys = [
+            a * self.x + b + np.random.normal(scale=0.1, size=(length,)) for _ in self.label_names
+        ]
+        self.ys = [y.astype(np.float32) for y in self.ys]
+
+    def __len__(self):
+        return self.length
+
+    def __getitem__(self, i):
+        result = {name: y[i] for name, y in zip(self.label_names, self.ys)}
+        result["input_x"] = self.x[i]
+        return result
+
+
+@pytest.mark.integrations
+class TestHuggingFace(BaseE2ETest):
+    @property
+    def _trainer_default_attributes(self):
+        model = RegressionModel(initial_a=2, initial_b=3)
+        train_dataset = RegressionDataset(length=32)
+        validation_dataset = RegressionDataset(length=16)
+
+        train_args = TrainingArguments(
+            "model", report_to=[], num_train_epochs=1000, learning_rate=0.5
+        )
+
+        return {
+            "model": model,
+            "args": train_args,
+            "train_dataset": train_dataset,
+            "eval_dataset": validation_dataset,
+        }
+
+    def test_every_train_should_create_new_run(self, environment):
+        common_tag = fake.nic_handle()
+        project = get_project(name=environment.project, api_token=environment.user_token)
+
+        trainer = Trainer(
+            **self._trainer_default_attributes,
```

Comment:
I don't get idea of adding `**` in prefix of function, have you forgotten `()` to call it?

Reply:
`_trainer_default_attributes` is property so `self._trainer_default_attributes` is proper call (output is dictionary of Trainer attributes)

Reply:
Right...